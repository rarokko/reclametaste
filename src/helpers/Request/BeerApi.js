import axios from 'axios';

class BeerApi {

  getBeers(keyword = "", filterName, page=1) {
    keyword = keyword !== "" ? `&${filterName}=${keyword}` : "";
    return axios.get(`https://api.punkapi.com/v2/beers?page=${page}&per_page=20${keyword}`);
  };
};

var beerApi = new BeerApi();
export default beerApi;
