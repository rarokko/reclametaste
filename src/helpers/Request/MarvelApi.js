import axios from 'axios';

class MarvelApi {

  constructor() {
    this.PUBLIC_API_KEY = 'fc84026e09ba6b818abec750c6c5868f';
  }

  getCharacters(keyword="", offset=0) {
    keyword = keyword !== "" ? `&nameStartsWith=${keyword}` : "";
    offset = offset !== 0 ? `&offset=${offset}` : "";
    return axios.get(`https://gateway.marvel.com/v1/public/characters?apikey=${this.PUBLIC_API_KEY}${keyword}${offset}`);
  };
  
  getComics(keyword="", offset=0) {
    keyword = keyword !== "" ? `&titleStartsWith=${keyword}` : "";
    offset = offset !== 0 ? `&offset=${offset}` : "";
    return axios.get(`https://gateway.marvel.com/v1/public/comics?apikey=${this.PUBLIC_API_KEY}${keyword}${offset}`);
  };
};

var marvelApi = new MarvelApi();
export default marvelApi;
