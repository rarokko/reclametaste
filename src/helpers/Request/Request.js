import fire from '../Firebase/Firebase';

class RequestApi {

    registerUser(email, password) {
        return fire.auth().createUserWithEmailAndPassword(email, password);
    }

    loginUser(email, password) {
        return fire.auth().signInWithEmailAndPassword(email, password)
    }

    logoutUser(email, password) {
        return fire.auth().signOut();
    }

    checkUserSession(email, password) {
        return fire.auth().currentUser;
    }

    getUserLikes(type, ) {
        const userId = fire.auth().currentUser.uid;
        return fire.database().ref(`${type}Likes/${userId}`).once("value");
    }

    toogleUserLike(type, item, status) {
        const userId = fire.auth().currentUser.uid;

        if (status === true) {
            return fire.database().ref(`${type}Likes/${userId}/${item.id}`).set({
                like: status,
                item: item
            });
        } else {
            return fire.database().ref(`${type}Likes/${userId}/${item.id}`).remove();
        }
    }
}

const Request = new RequestApi();
export default Request;