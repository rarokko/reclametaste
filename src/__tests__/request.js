import React from 'react';
import Request from '../helpers/Request/Request';

describe('Request', () => {
  it('the same userid on login', async () => {

    let response = await Request.loginUser('rarokko@msn.com', '123456');
    expect(response.user.uidu).toMatchSnapshot();

  });

  it('check user session', async () => {

    let response = await Request.checkUserSession();
    expect(response).not.toBe(null);

  });

  it('get marvel likes', async () => {

    let response = await Request.getUserLikes('marvel');
    expect(response).not.toBe(null);

  });

  it('insert marvel likes', async () => {

    let response = await Request.toogleUserLike('marvel', 1011334, true);
    expect(response).toMatchSnapshot();

  });

  it('remove marvel likes', async () => {

    let response = await Request.toogleUserLike('marvel', 1011334, false);
    expect(response).toMatchSnapshot();

  });

  it('get beer likes', async () => {

    let response = await Request.getUserLikes('beer');
    expect(response).not.toBe(null);

  });

  it('insert beer likes', async () => {

    let response = await Request.toogleUserLike('beer', 1011334, true);
    expect(response).toMatchSnapshot();

  });

  it('remove beer likes', async () => {

    let response = await Request.toogleUserLike('beer', 1011334, false);
    expect(response).toMatchSnapshot();

  });
});