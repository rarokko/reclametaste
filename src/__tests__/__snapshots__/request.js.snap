// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request insert beer likes 1`] = `undefined`;

exports[`Request insert marvel likes 1`] = `undefined`;

exports[`Request remove beer likes 1`] = `undefined`;

exports[`Request remove marvel likes 1`] = `undefined`;

exports[`Request the same userid on login 1`] = `undefined`;
